### PR TITLE
Added Proxy Support

### PR DESCRIPTION
**Pull request by Rajala1404**
By using the http_proxy Package i was able to implement Proxy Support. Also works with Auto Config.

Tested with Android 14 on Galaxy S23+ (Auto Config - PAC and Manual)

Closes #228 and #221

### CODE REVIEW

File: app/lib/main.dart [hunk 1]
```diff
@@ -96,6 +97,10 @@ void main() async {
     ThemeModeNotifier.init();
     ColorModeNotifier.init();
 
+    WidgetsFlutterBinding.ensureInitialized();
```
Comment (alessioC42) [inline app/lib/main.dart:100]: This line can be removed. It was already called on line 29

**Review by alessioC42 — CHANGES_REQUESTED** [review]
That's a bummer, I didn't think to override the global instance, only the separate dio instance. xD

Still have to test it, but looks good as of now.

**Comment by Rajala1404** [comment]
Before you merge I need to Revert my Commit (Theme) that wasn't meant for this Pull Request.

**Comment by Rajala1404** [comment]
Is now ready to merge

**Review by alessioC42 — APPROVED** [review]
Looks good. The only problem I see is that `http_proxy` uses deprecated API's, but that should be fixable with a PR on this project.
 I think I will look into it. Might be a small step to learn java

**Pull request merged**